src/App.js

```diff
@@ -1,17 +1,201 @@
+import { WelcomeMsg } from "./WelcomeMsg.js";
+import { Instruction } from "./Instruction.js";
+import { InputForm } from "./InputForm.js";
+import { PlayForm } from "./PlayForm.js";
+import { LeaderBoard } from "./LeaderBoard.js";
+import { DiceImage } from "./DiceImage.js";
+import { rollDice, nextPlayer, nextRound } from "./utils.js";
+import { isLastPlayer, determineWinner, soundPlay } from "./utils.js";
+
+// define the audio clips used in this APP
+const audioClips = [
+  {
+    sound: "http://cd.textfiles.com/itcontinues/WIN/YTB22/RATTLE2.WAV",
+    label: "Dice Shake",
+  },
+  {
+    sound:
+      "https://soundbible.com/mp3/Short_triumphal_fanfare-John_Stracke-815794903.mp3",
+    label: "Victory sound",
+  },
+  {
+    sound:
+      "https://soundbible.com/mp3/jack_in_the_box-Mike_Koenig-710345321.mp3",
+    label: "Game music",
+  },
+];
 
 class App extends React.Component {
+  constructor(props) {
+    // Always call super with props in constructor to initialise parent class
+    super(props);
+    this.state = {
+      gameStarted: false,
+      numberOfPlayers: 2, // form input state
+      numberOfRounds: 3,
+      userRollDice: false,
+      userOrderDice: false,
+      userScores: [],
+      currentPlayer: 0,
+      currentRound: 0,
+      playerDiceRolls: [0, 0],
+      currentPlayerScore: 0,
+      diceOrder: 1,
+      audio: new Audio(audioClips[2].sound), // set background music
+      isMusicOn: false, // background music is initially OFF
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleContinue = this.handleContinue.bind(this);
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({
+      [name]: parseInt(value),
+    });
+  }
+
+  handleSubmit(event) {
+    const { name, value } = event.target;
+    event.preventDefault();
+    this.state.audio.play(); // start playing background music
+    this.state.audio.loop = true;
+
+    this.setState((previousState) => ({
+      [name]: value,
+      userScores: Array(this.state.numberOfPlayers).fill(0),
+      gameStarted: true,
+      isMusicOn: true,
+    }));
+  }
+
+  handleContinue(event) {
+    event.preventDefault();
+    if (!this.state.userRollDice) {
+      soundPlay(audioClips[0].sound, false, 1); // play Dice shake - louder
+      this.setState((previousState) => ({
+        currentPlayer: nextPlayer(
+          previousState.currentPlayer,
+          this.state.numberOfPlayers
+        ),
+        playerDiceRolls: [rollDice(), rollDice()],
+        userRollDice: !previousState.userRollDice,
+        userOrderDice: false,
+      }));
+    } else {
+      let userScore =
+        this.state.diceOrder === 1
+          ? this.state.playerDiceRolls[0] * 10 + this.state.playerDiceRolls[1]
+          : this.state.playerDiceRolls[1] * 10 + this.state.playerDiceRolls[0];
+      const newScores = this.state.userScores.slice(); //copy the array
```

Comment:
We could just do this
```suggestion
      const newScores = [...this.state.userScores]
```